sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -571,6 +580,62 @@ public static <T> T getResultWithTimeout(Future<T> future, Duration timeout)
         return ImplUtils.getResultWithTimeout(future, timeout.toMillis());
     }
 
+    /**
+     * Helper method that safely adds a {@link Runtime#addShutdownHook(Thread)} to the JVM that will close the
+     * {@code executorService} when the JVM is shutting down.
+     * <p>
+     * {@link Runtime#addShutdownHook(Thread)} checks for security privileges and will throw an exception if the proper
+     * security isn't available. So, if running with a security manager, setting
+     * {@code AZURE_CORE_UTILS_SHUTDOWN_HOOK_ACCESS_HELPER} to true will have this method use access controller to add
+     * the shutdown hook with privileged permissions.
+     * <p>
+     * If {@code executorService} is null, no shutdown hook will be added and this method will return null.
+     * <p>
+     * The {@code shutdownTimeout} is the amount of time to wait for the {@code executorService} to shutdown. If the
+     * {@code executorService} doesn't shutdown within half the timeout, it will be forcefully shutdown.
+     *
+     * @param executorService The {@link ExecutorService} to shutdown when the JVM is shutting down.
+     * @param shutdownTimeout The amount of time to wait for the {@code executorService} to shutdown.
+     * @return The {@code executorService} that was passed in.
+     * @throws NullPointerException If {@code shutdownTimeout} is null.
+     * @throws IllegalArgumentException If {@code shutdownTimeout} is zero or negative.
+     */
+    @SuppressWarnings({"deprecation", "removal"})
+    public static ExecutorService addShutdownHookSafely(ExecutorService executorService, Duration shutdownTimeout) {
+        if (executorService == null) {
+            return null;
+        }
+        Objects.requireNonNull(shutdownTimeout, "'shutdownTimeout' cannot be null.");
+        if (shutdownTimeout.isZero() || shutdownTimeout.isNegative()) {
+            throw new IllegalArgumentException("'shutdownTimeout' must be a non-zero positive duration.");
+        }
+
+        long timeoutNanos = shutdownTimeout.toNanos();
+        Thread shutdownThread = new Thread(() -> {
+            try {
+                executorService.shutdown();
+                if (!executorService.awaitTermination(timeoutNanos / 2, TimeUnit.NANOSECONDS)) {
+                    executorService.shutdownNow();
+                    executorService.awaitTermination(timeoutNanos / 2, TimeUnit.NANOSECONDS);
+                }
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
```

Comment:
Should we log the exception here?

Reply:
We don't in any place doing this currently, given this is happening during shutdown maybe it's best we don't?